src/HealthChecks/HealthChecks/src/DefaultHealthCheckService.cs

```diff
@@ -86,6 +92,19 @@ internal class DefaultHealthCheckService : HealthCheckService
                             Log.HealthCheckData(_logger, registration, entry);
                         }
 
+                        catch (OperationCanceledException ex) when (!cancellationToken.IsCancellationRequested)
+                        {
+                            var duration = stopwatch.GetElapsedTime();
+                            entry = new HealthReportEntry(
+                                status: HealthStatus.Unhealthy,
+                                description: "A timeout occured while running check.",
+                                duration: duration,
+                                exception: ex,
+                                data: null);
+
+                            Log.HealthCheckError(_logger, registration, ex, duration);
+                        }
+
```

Comment:
nit: update the comment at L108 to say `Allow cancellation to propagate if it's not a timeout.`